sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1733,6 +1762,11 @@ object GpuOverrides extends Logging {
         TypeSig.TIMESTAMP, TypeSig.TIMESTAMP),
       (second, conf, p, r) => new UnaryExprMeta[Second](second, conf, p, r) {
 
+       override def tagExprForGpu(): Unit = {
```

Comment:
Could we try and have a TimeZoneAwareExprMeta, or something similar that makes it super simple to do this? We might even be able to back it into ExprMeta itself, just by checking if the class that this wraps is also `TimeZoneAware`.

Reply:
Maybe wrap the check in a method and override it whenever a function starts supporting alternate timezones.

Reply:
Looks like GpuCast will be a first exception to this idea: https://github.com/NVIDIA/spark-rapids/issues/6835

Reply:
I'm testing all the existing test cases with adding non-UTC time zone config to identify all the failed cases:

https://github.com/NVIDIA/spark-rapids/blob/v23.10.0/integration_tests/src/main/python/spark_session.py#L68-L74

```
def _set_all_confs(conf):
    _spark.conf.set("spark.sql.session.timeZone": "+08:00")
```

Then I'll update the failed cases.